internal/controllers/core/cluster/reconciler.go

```diff
@@ -103,6 +104,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		}
 	}
 
+	if !hasConnection {
+		r.reportConnectionEvent(ctx, conn)
```

Comment:
this is going to report on every re-reconcile.

can you put this in `maybeUpdateStatus` and only update on state change?